### PR TITLE
Create Fibonachi.java

### PR DESCRIPTION
**Pull request by nikitoshom**

### CODE REVIEW

File: Fibonachi.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import java.util.Scanner;
+
+public class Fibonachi {
+  public static void main (String[] args) {
+    System.out.println("Please enter a number:");
+    Scanner number = new Scanner(System.in);
+    String string = number.next();
+    int number_int = Integer.parseInt(string);
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: codestyle violation

File: Fibonachi.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    int firstNumber = 1;
+    int secondNumber = 1;
+    int thirdNumber = 1;
+    int countNumberFebonacci = 0;
+    int numberForTest = 0; 
+    boolean resultOfTest = false;
+    if (number_int == 0 || number_int == 1 || number_int == -1) {
+      System.out.println("It's number of sequence Febonachi");
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: return from method here and drop off else block in that case

File: Fibonachi.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    } else {
+      if (number_int < 0) {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: You cannot do it as user explicitly said he want to check negative number. He doesn't want to check |negative number|

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: Fibonachi.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import java.util.Scanner;
+
+public class Fibonachi {
+  public int number_int;
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: why public? naming convention violation

File: Fibonachi.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  public void InputNumber() {
+    System.out.println("Please enter a number:");
+    Scanner number = new Scanner(System.in);
+    String stringWithNumber = number.next();
+    number_int = Integer.parseInt(stringWithNumber);
+  }
+  public boolean TestFibonachi() {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: pleas add blank line before method name

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: Fibonachi.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import java.util.Scanner;
+
+public class Fibonachi {
+  public int InputNumber() {
+    System.out.println("Please enter a number:");
+    Scanner number = new Scanner(System.in);
+    String stringWithNumber = number.next();
+    int number_int = Integer.parseInt(stringWithNumber);
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: again, naming convention violation. Please fix it to proceed with the review.

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: Fibonachi.java (hunk 6 above)
Comment (nikitoshom) [reply Fibonachi.java]: Please, help. I don't understand about what kind of error is it? In which line?

File: Fibonachi.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import java.util.Scanner;
+
+public class Fibonachi {
+  public int InputNumber() {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: javadoc missed

File: Fibonachi.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    System.out.println("Please enter a number:");
+    Scanner number = new Scanner(System.in);
+    String stringWithNumber = number.next();
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: why not nextInt()?

File: Fibonachi.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    return numberInt;
+  }
+  
+  public boolean TestFibonachi(int numberInt) {
+    int firstNumber = 1;
+    int secondNumber = 1;
+    int thirdNumber = 1;
+    boolean resultOfTest = false;
+    if (numberInt == 0 || numberInt == 1) {
+      resultOfTest = true;
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: avoid using any flags, just return true here

File: Fibonachi.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  public int InputNumber() {
+    System.out.println("Please enter a number:");
+    Scanner number = new Scanner(System.in);
+    String stringWithNumber = number.next();
+    int numberInt = Integer.parseInt(stringWithNumber);
+    return numberInt;
+  }
+  
+  public boolean TestFibonachi(int numberInt) {
+    int firstNumber = 1;
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: declare variable as close to first usage as possible

File: Fibonachi.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    boolean resultOfTest = false;
+    if (numberInt == 0 || numberInt == 1) {
+      resultOfTest = true;
+    } else {   
+      for (int i = 1; i <= numberInt; i++) {
+        secondNumber = firstNumber;
+        firstNumber = thirdNumber;
+        thirdNumber = firstNumber + secondNumber;			
+        if (thirdNumber == numberInt) {
+          resultOfTest = true;
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: return true;

File: Fibonachi.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+          break;
+        }			
+      }	
+    }	
+    return resultOfTest;
+  }
+  
+  public void OutputResult (Boolean resultOfTest) {
+    if (resultOfTest == true) {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: if (result)

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: Fibonachi.java [hunk 13]
```diff
@@ -0,0 +1,45 @@
+import java.util.Scanner;
+
+/**
+ * This program checks whether the number entered from the keyboard is a nonnegative number of sequence Fibonachi.
+ */
+public class Fibonachi {
+  public int InputNumber() {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: javadoc comment missed, naming convention violation (codestyle 5.2.3)

File: Fibonachi.java [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    System.out.println("Please enter a number:");
+    Scanner number = new Scanner(System.in);
+    int numberInt = number.nextInt();
+    return numberInt;
+  }
+  
+  public boolean TestFibonachi(int numberInt) {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: javadoc comment missed, naming convention violation (codestyle 5.2.3)

File: Fibonachi.java [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+        thirdNumber = firstNumber + secondNumber;			
+        if (thirdNumber == numberInt) {
+          return true;
+        }			
+      }	
+    }	
+    return false;
+  }
+  
+  public void OutputResult (Boolean resultOfTest) {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: javadoc comment missed, naming convention violation (codestyle 5.2.3)

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
Javadoc violations: neither arguments nor return value are described

File: Fibonachi.java [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+   * This method test whether the entered number is the number of sequence Fibonachi. 
+   * Method get number as a parameter which we want to test, and return true or false depending on the passage of the test.
+   * 
+   * @param number which we entered from keyboard
+   * @return true if number is a number of sequence of Fibonachi and false if don't
+   */ 
+  public boolean testFibonachi(int numberInt) {
+    if (numberInt == 0 || numberInt == 1) {
+      return true;
+    } else {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: Useless else (as you have return in "if" block)

File: Fibonachi.java [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+    }	
+    return false;
+  }
+  
+  /**
+   * This method outputs the result of this program. 
+   * 
+   * @param boolean result of previos method (testFibonachi()).
+   */ 
+  public void outputResult (Boolean resultOfTest) {
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: non-appropriate argument name

File: Fibonachi.java [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+        }			
+      }	
+    }	
+    return false;
+  }
+  
+  /**
+   * This method outputs the result of this program. 
+   * 
+   * @param boolean result of previos method (testFibonachi()).
```
Comment (AlexanderSoroka) [inline Fibonachi.java]: Who guarantees that previous method always is  testFibonachi?

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

**Review by AlexanderSoroka — APPROVED** [review]

**Review by AlexanderSoroka — APPROVED** [review]

**Pull request merged**